src/App.jsx

```diff
@@ -8,12 +8,14 @@ import ArticlesList from "./components/ArticlesList";
 import { getTopics } from "./utils/axiosFunctions";
 import SingleArticle from "./components/SingleArticle";
 import FilterBar from "./components/FilterBar";
+import ErrorHandler from "./components/ErrorHandler";
 
 function App() {
   const [allArticles, setAllArticles] = useState([]);
   const [loading, setLoading] = useState(true);
   const [usernameLoggedIn, setUsernameLoggedIn] = useState("happyamy2016");
   const [allTopics, setAllTopics] = useState([]);
+  const [isError, setIsError] = useState([false, null]);
```

Comment:
I'm not sure about having the error state as an array. By definition, null is a falsy value so you could just set error to be null or to be an error object - this is perfectly valid as you can just check `if (isError)` in your components that use this state.